src/lib/mongoBackend/mongoRegistrationGet.cpp

```diff
@@ -0,0 +1,325 @@
+
+    if (ce.hasField(REG_ENTITY_ISPATTERN))
+    {
+      std::string isPattern = getStringFieldF(ce, REG_ENTITY_ISPATTERN);
+
+      if (isPattern == "true")
+      {
+        entity.idPattern = getStringFieldF(ce, REG_ENTITY_ID);
+      }
+      else
+      {
+        entity.id = getStringFieldF(ce, REG_ENTITY_ID);
+      }
+    }
+    else
+    {
+      entity.id = getStringFieldF(ce, REG_ENTITY_ID);
+    }
+    
+    if (ce.hasField(REG_ENTITY_ISTYPEPATTERN))
+    {
+      std::string isPattern = getStringFieldF(ce, REG_ENTITY_ISTYPEPATTERN);
+
+      if (isPattern == "true")
+      {
+        entity.typePattern = getStringFieldF(ce, REG_ENTITY_TYPE);
+      }
+      else
+      {
+        entity.type = getStringFieldF(ce, REG_ENTITY_TYPE);
+      }
+    }
+    else
+    {
+      entity.type = getStringFieldF(ce, REG_ENTITY_TYPE);
+    }
+
+    regP->dataProvided.entities.push_back(entity);
+  }
+
+
+  //
+  // Attributes
+  //
+  std::vector<mongo::BSONElement> dbAttributeV = getFieldF(cr0, REG_ATTRS).Array();
+  for (unsigned int ix = 0; ix < dbAttributeV.size(); ++ix)
+  {
+    mongo::BSONObj  aobj     = dbAttributeV[ix].embeddedObject();
+    std::string     attrName = getStringFieldF(aobj, REG_ATTRS_NAME);
+
+    if (attrName != "")
+    {
+      regP->dataProvided.attributes.push_back(attrName);
+    }
+  }
+
+  //
+  // Providing application
+  //
+  setProvider(regP, cr0);
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* setExpires -
+*/
+static void setExpires(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->expires = (r.hasField(REG_EXPIRATION))? getIntFieldF(r, REG_EXPIRATION) : -1;
+}
+
+
+
+/* ****************************************************************************
+*
+* setStatus -
+*/
+static void setStatus(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->status = (r.hasField(REG_STATUS))? getStringFieldF(r, REG_STATUS): "";
+}
+
+
+
+/* ****************************************************************************
+*
+* setForwardingInformation -
+*/
+static void setForwardingInformation(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+}
+
+
+
+/* ****************************************************************************
+*
+* mongoRegistrationGet - 
+*/
+void mongoRegistrationGet
+(
+  ngsiv2::Registration*  regP,
+  const std::string&     regId,
+  const std::string&     tenant,
+  const std::string&     servicePath,
+  OrionError*            oeP
+)
+{
+  bool         reqSemTaken = false;
+  std::string  err;
+  mongo::OID   oid;
+  StatusCode   sc;
+
+  if (safeGetRegId(regId, &oid, &sc) == false)
+  {
+    oeP->fill(sc);
+    return;
+  }
+
+  reqSemTake(__FUNCTION__, "Mongo Get Registration", SemReadOp, &reqSemTaken);
+
+  LM_T(LmtMongo, ("Mongo Get Registration"));
+
+  std::auto_ptr<mongo::DBClientCursor>  cursor;
+  mongo::BSONObj                        q;
+
+  if (servicePath.empty())
```

Comment:
Looking to how this implemented in subscriptions:

```
BSONObj q     = BSON("_id" << oid);
```

(In other words, SP is not taken into account in the query)

So my nice theory about how SP works in subs (and should work in regs) exposed in https://github.com/telefonicaid/fiware-orion/issues/3051 colapses down :D

Thus, I propose the following:

1. To create an issue about "how SP should be interpreted in retrieve operations for subscriptions and registrations" (expaning #3051 or creating a new one)
2. Implement this by the moment with `q = BSON("_id" << oid);` as subscription are, pending to have a final decision of what to do in the issue.

Reply:
Well, not completely collapsed ;)... I think is stills making sense but for other operation.

It seems that SP is taken into account in "getting all" operation (i.e. GET /v2/subscrpitions, without id):

```
  if (!servicePath.empty() && servicePath != "/#")
  {
    q = Query(BSON(CSUB_SERVICE_PATH << servicePath));
  }
  else
  {
    q = Query();
  }
```

Maybe a good argument for not taking into account SP in GET /v2/subscriptions/id and GET /v2/registrations/id is that id univocally identifies the subscription/registration in that case, i.e. the "primary key" of the object doesn't include SP (as entities do).

Reply:
Yeah, it gets a bit tricky with the service path and the unique reg id ...
If you have the reg id ... should we even care about the service path?
I think I would vote against that, but open for discussion.

Let's decide something asap and I'll change this accordingly.

And yes, when more than one registration (without any reg id (GET /v2/registrations), the service path makes a lot of sense. Same for subs, of course.

Reply:
> Let's decide something asap and I'll change this accordingly, if necessary.

For this PR, ignore SP in the query and use just "q = BSON("_id" << oid);"

I'll take care of adjusting #3051 and add a piece of information about this.

Reply:
Fixed in bd588f384